src/lib.rs

```diff
@@ -1051,15 +1160,15 @@ struct ServerInfo {
     connect_urls: Vec<String>,
 }
 
-#[inline(always)]
-fn default_false() -> bool {
+const fn default_false() -> bool {
     false
 }
-#[inline(always)]
+
+#[inline]
 fn default_empty() -> String {
     "".to_string()
 }
-#[inline(always)]
-fn default_no_urls() -> Vec<String> {
-    vec![]
+
+const fn default_no_urls() -> Vec<String> {
+    Vec::new()
```

Comment:
Fun note: `Vec::new()` is a `const fn` because it does not actually allocate anything until you push something into the `Vec`.

Reply:
nice!

Reply:
I knew that but did n't think to promote to fn level.